### ISSUE TITLE
libressl fails to build after 2024-04-01

### ISSUE THREAD

**Issue by bmwiedemann** [libressl/portable#1018]
While working on [reproducible builds](https://reproducible-builds.org/) for [openSUSE](https://en.opensuse.org/openSUSE:Reproducible_Builds), I found that
[our `libressl-3.8.3` package](https://code.opensuse.org/package/libressl/blob/master/f/libressl.spec#_117) fails to build after 2024-04-01, because the `tlstest.sh` test fails because `libressl-3.8.3/tests/client.pem` and `server.pem` expire 2024-04-01T14:45:11 UTC. There is also a `ca.pem` expiring a month later, but not sure if that has an impact.

Steps to reproduce (on Debian or openSUSE)
```bash
osc checkout openSUSE:Factory/libressl && cd $_
osc build --vm-type=kvm --noservice --clean --build-opt=--vm-custom-opt="-rtc base=2034-04-02T00:00:00" standard
```

Background:
As part of my work on reproducible builds for openSUSE, I check that software still gives identical build results in the future.
The usual offset is +16 years, because that is how long I expect some software will be used in some places.
This showed up failing tests in our package build.

**Comment by botovq** [comment]
Thanks a lot for the report - this is bad. We will fix this as soon as
we can.

**Comment by joshuasing** [comment]
@bmwiedemann Thank you for this report! This has been resolved in v3.9.1 and v3.8.4.
https://github.com/libressl/portable/commit/86e4965d7f20c3a6afc41d95590c9f6abb4fe788

**Comment by bmwiedemann** [comment]
I found that new test-expiry date with v3.8.4 is 2029-05-08
